### PR TITLE
Update resource naming to allow for a much wider range of server names

### PR DESCRIPTION
**Pull request by jsonar-cpapke**

### CODE REVIEW

File: modules/azurerm/sonar-base-instance/main.tf [hunk 1]
```diff
@@ -89,15 +89,16 @@ resource "azurerm_linux_virtual_machine" "dsf_base_instance" {
...
   resource_group_name = var.resource_group.name
   location            = var.resource_group.location
 }
 
-data "azurerm_subscription" "subscription" {
-}
+data "azurerm_subscription" "subscription" {}
 
 resource "azurerm_role_assignment" "dsf_base_storage_role_assignment" {
+  count                = var.binaries_location.az_resource_group != "" ? 1 : 0
```
Comment (eytannnaim) [inline modules/azurerm/sonar-base-instance/main.tf:101]: Wouldn't it be more right to pass null in binaries_location?

Comment (jsonar-cpapke) [reply modules/azurerm/sonar-base-instance/main.tf:101]: Currently it defaults to empty string in https://github.com/imperva/dsfkit/pull/335

Comment (eytannnaim) [reply modules/azurerm/sonar-base-instance/main.tf:101]: ok

**Review by eytannnaim — APPROVED** [review]

**Comment by jsonar-cpapke** [comment]
The main note is that while this does allow a much larger range of names, anyone using dsfkit already would get a complete recreate of their servers as all the vault names will change.

**Comment by eytannnaim** [comment]
That's ok.

**Pull request merged**